setup-mux.sh

```diff
@@ -0,0 +1,377 @@
+    # test/dev env - see if we have a KUBECONFIG
+    if [ -n "${KUBECONFIG:-}" ] ; then
+        MASTER_CONFIG_DIR=`dirname $KUBECONFIG`
+    fi
+fi
+
+if [ ! -d $MASTER_CONFIG_DIR ] ; then
+    # get from openshift server ps
+    # e.g.
+    #    root      2477  2471 30 04:48 ?        04:03:58 /data/src/github.com/openshift/origin/_output/local/bin/linux/amd64/openshift start --loglevel=4 --logspec=*importer=5 --latest-images=false --node-config=/tmp/openshift/origin-aggregated-logging//openshift.local.config/node-192.168.78.2/node-config.yaml --master-config=/tmp/openshift/origin-aggregated-logging//openshift.local.config/master/master-config.yaml
+    MASTER_CONFIG_DIR=`ps -ef|grep -v awk|awk '/openshift.*master-config=/ {print gensub(/^.*--master-config=(\/.*)\/master-config.yaml.*$/, "\\\1", 1)}'|head -1`
+fi
+
+if [ ! -f $MASTER_CONFIG_DIR/ca.key ] ; then
+    err could not find the openshift ca key needed to create the mux server cert
+    err Check your permissions - you may need to run this script as root
+    err Otherwise, please specify MASTER_CONFIG_DIR correctly and re-run this script
+    exit 1
+fi
+
+cacert=$MASTER_CONFIG_DIR/ca.crt
+cakey=$MASTER_CONFIG_DIR/ca.key
+caser=$MASTER_CONFIG_DIR/ca.serial.txt
+
+workdir=`mktemp -d`
+
+info cleanup old mux configuration, if any . . .
+oc delete -n logging dc logging-mux > /dev/null 2>&1 || :
+oc delete -n logging secret logging-mux  > /dev/null 2>&1|| :
+oc delete -n logging configmap logging-mux  > /dev/null 2>&1|| :
+oc delete -n logging service logging-mux  > /dev/null 2>&1|| :
+oc delete -n logging route logging-mux  > /dev/null 2>&1|| :
+oc delete project mux-undefined  > /dev/null 2>&1 && sleep 10 || : # give it some time before recreating
+
+# generate mux server cert/key
+info generate mux server cert, key
+openshift admin ca create-server-cert  \
+          --key=$workdir/mux.key \
+          --cert=$workdir/mux.crt \
+          --hostnames=mux,$MUX_HOST \
+          --signer-cert=$cacert --signer-key=$cakey --signer-serial=$caser
+
+# generate mux shared_key
+info generate mux shared key
+openssl rand -base64 48 > "$workdir/mux-shared-key"
+
+# add secret for mux
+info create secret for mux
+oc secrets -n logging new logging-mux \
+   mux-key=$workdir/mux.key mux-cert=$workdir/mux.crt \
+   mux-shared-key=$workdir/mux-shared-key mux-ca=$cacert > /dev/null
+
+# add mux secret to fluentd service account
+info add mux secret to fluentd service account
+oc secrets -n logging add serviceaccount/aggregated-logging-fluentd \
+   logging-fluentd logging-mux > /dev/null
+
+# add namespace for records from unknown namespaces
+info add namespace [mux-undefined] for records from unknown namespaces
+oadm new-project mux-undefined --node-selector='' > /dev/null
+
+# add namespaces for projects
+for ns in ${MUX_NAMESPACES:-} ; do
+    info adding namespace [$ns]
+    oadm new-project $ns --node-selector='' > /dev/null
+done
+
+# # allow externalIPs in services
```

Comment:
It is open by default in origin 1.5/ocp 3.5.  But it appears for 1.6/3.6 that you have to explicitly configure OpenShift to allow the use of external IPs.  Hence the comment.  I expect this will be handled in the ansible code - there is probably some configuration parameter that can be set in the inventory and/or vars.yaml to specify the list of IP addresses/CIDRs to set.